### PR TITLE
add index to transactions controller

### PR DESCRIPTION
**Pull request by se3000**
Sounds like we still have reason to keep the transaction attempts controller around for debugging, so this PR just introduces the transactions index, and leaves the transaction attempts command.

Also, cleaned up JSON marshaling in the TX presenter and tried to abstract some of the duplicate logic for paginated results in controllers.

### CODE REVIEW

File: web/helpers.go [hunk 1]
```diff
@@ -26,3 +29,23 @@ func publicError(c *gin.Context, statusCode int, err error) {
...
+func pagedResponse(
+	c *gin.Context,
+	name string,
+	size int,
+	page int,
+	resource interface{},
+	count int,
+	err error,
+) {
+	if err == orm.ErrorNotFound {
```
Comment (j16r) [inline web/helpers.go]: Does 404 make sense for an index endpoint? I would expect an empty list.

File: web/tx_attempts_controller_test.go [hunk 2]
```diff
@@ -34,13 +35,15 @@ func TestTxAttemptsController_Index_Success(t *testing.T) {
...
-	resp, cleanup := client.Get("/v2/txattempts?size=2")
+	resp, cleanup := client.Get("/v2/tx_attempts?size=2")
 	defer cleanup()
 	cltest.AssertServerResponse(t, resp, 200)
 
 	var links jsonapi.Links
 	var attempts []models.TxAttempt
-	err = web.ParsePaginatedResponse(cltest.ParseResponseBody(resp), &attempts, &links)
+	body := cltest.ParseResponseBody(resp)
+	fmt.Println("$$$$$$$$$$", string(body))
```
Comment (j16r) [inline web/tx_attempts_controller_test.go]: Debug?

File: web/helpers.go (hunk 1 above)
Comment (se3000) [reply web/helpers.go]: Doesn't make sense. Good catch.

**Comment by se3000** [comment]
On the CC "too many parameters" issue, we could easily introduce a type to encapsulate those parameters, but I'd like to see some support on this pattern before putting more effort into it.

File: web/helpers.go [hunk 3]
```diff
@@ -26,3 +29,34 @@ func publicError(c *gin.Context, statusCode int, err error) {
...
+func paginatedResponse(
+	c *gin.Context,
+	name string,
+	size int,
+	page int,
+	resource interface{},
+	count int,
+	err error,
+) {
+	if err == orm.ErrorNotFound {
```
Comment (j16r) [inline web/helpers.go]: Still 404? Shouldn't be 2xx?

Comment (j16r) [reply web/helpers.go]: Sounds good.

File: store/presenters/presenters.go [hunk 4]
```diff
@@ -488,7 +489,34 @@ func (a NewAccount) GetName() string {
...
+	SentAt    string         `json:"sentAt"`
+	To        common.Address `json:"to"`
+	Value     string         `json:"value"`
+}
+
+// NewTx builds a transaction presenter.
+func NewTx(tx *models.Tx) Tx {
+	return Tx{
+		Confirmed: tx.Confirmed,
+		Data:      hexutil.Bytes(tx.Data),
```
Comment (j16r) [inline store/presenters/presenters.go]: Are these "casts" needed? Both structs should have the same type.

Comment (se3000) [reply store/presenters/presenters.go]: You can actually see what the JSON marshaling and unmarshaling look like if we don't make them separate types in [the second commit in this PR](https://github.com/smartcontractkit/chainlink/pull/947/commits/919259ebf712bc6a67c4b3aff49ec5bcb132303c#diff-1ff93c0482d8d63d4388a5145fca5250L525). There's a lot more coercion logic split between the two. By using different types in the presenters, those types have their own JSON marshaling logic and the Tx presenter is much simpler. Hat tip to @dimroc for the suggestion, I personally think it's cleaner this way.

Comment (j16r) [reply store/presenters/presenters.go]: Oh, then they do have different types.

**Pull request merged**